### PR TITLE
Openssl 3.0.15

### PR DESCRIPTION
**Pull request by liyi77**
# Description
https://bugzilla.tianocore.org/show_bug.cgi?id=4842

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [x] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - fix CVE-2024-6119: https://www.cve.org/CVERecord?id=CVE-2024-6119
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Unit test of CryptoPkg.

## Integration Instructions

NA

### CODE REVIEW

**Comment by liyi77** [comment]
Size impact between 3.0.9~3.0.15(no compressed):
No increase in Pei driver [Only AES, SHA, SM3]
65KB increase in FullDxe driver [All]

File: CryptoPkg/Library/OpensslLib/OpensslGen/include/openssl/bio.h [hunk 1]
```diff
@@ -867,7 +867,7 @@ int BIO_meth_set_puts(BIO_METHOD *biom,
                       int (*puts) (BIO *, const char *));
 int (*BIO_meth_get_gets(const BIO_METHOD *biom)) (BIO *, char *, int);
 int BIO_meth_set_gets(BIO_METHOD *biom,
-                      int (*gets) (BIO *, char *, int));
+                      int (*ossl_gets) (BIO *, char *, int));
```
Comment (AkechiShiro) [inline CryptoPkg/Library/OpensslLib/OpensslGen/include/openssl/bio.h:870]: Is there any reason this function got renamed ? Does this come from a change in openssl upstream if so could you let us know which change exactly ?

Comment (liyi77) [reply CryptoPkg/Library/OpensslLib/OpensslGen/include/openssl/bio.h:870]: Please refer to this: https://github.com/openssl/openssl/issues/22296

**Comment by liyi77** [comment]
Hi @jyao1 , could you take a look?

**Comment by jyao1** [comment]
@liyi77 , would you please provide the compressed size difference?

**Review by jyao1 — APPROVED** [review]

**Comment by liyi77** [comment]
> @liyi77 , would you please provide the compressed size difference?

there will be ~15KB increase after compressed.

**Pull request merged**